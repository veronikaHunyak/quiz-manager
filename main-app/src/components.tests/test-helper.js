export const mockQuiz = [
  {
    _id: "5eb14db7c8248d0f7cd4a279",
    quizKey: "quiz0",
    quizName: "First quiz test name",
    quizData: [
      {
        question: "Question 1",
        options: [{ A: "one" }, { B: "two" }, { C: "three" }],
        answer: "B",
      },
      {
        question: "question2",
        options: [{ A: "a" }, { B: "b" }, { C: "c" }, { D: "d" }],
        answer: "C",
      },
    ],
  },
];
