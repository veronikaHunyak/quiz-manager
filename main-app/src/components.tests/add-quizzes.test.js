import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import AddQuizzes from "../components/add-quizzes";
import { mockQuiz } from "./test-helper";

describe("AddQuizzes", () => {
  it("should render correctly", () => {
    const wrapper = shallow(<AddQuizzes />);
    wrapper.setState({ quiz: mockQuiz });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot when quizzes is populated", () => {
    const wrapper = shallow(<AddQuizzes />);
    wrapper.setState({ quizzes: mockQuiz });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
