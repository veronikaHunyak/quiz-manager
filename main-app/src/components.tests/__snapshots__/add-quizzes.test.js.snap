// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddQuizzes should match snapshot when quizzes is populated 1`] = `
<div>
  <h6
    className="container-fluid"
    style={
      Object {
        "margin": 10,
      }
    }
  >
    Hello, here you can create new quizzes
  </h6>
  <a
    href="/quizzes"
  >
    <button
      className="btn btn-primary"
      style={
        Object {
          "margin": "10px",
        }
      }
      type="button"
    >
      <ChevronBarLeft
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      View quizzes
    </button>
  </a>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "10px",
      }
    }
    type="button"
  >
    <PlusSquare
      color="currentColor"
      size="1em"
      style={
        Object {
          "marginRight": "10px",
        }
      }
    />
    Add a new Quiz
  </button>
  <br />
  <div
    className="container-fluid card"
    style={
      Object {
        "backgroundColor": "seashell",
        "marginBottom": "35px",
      }
    }
  >
    <form
      action=""
    >
      <div
        className="form-group"
        style={
          Object {
            "padding": "10px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-2"
          >
            <label
              for="email"
            >
              Quiz name:
            </label>
          </div>
          <div
            className="col-8"
          >
            <input
              className="form-control"
              id="quiz0"
              onChange={[Function]}
              placeholder="Enter quiz name"
              type="text"
            />
          </div>
          <div
            className="col-2"
          >
            <button
              className="btn btn-outline-danger"
              onClick={[Function]}
              style={
                Object {
                  "backgroundColor": "lavenderblush",
                }
              }
              type="button"
            >
              <PlusSquare
                color="currentColor"
                size="1em"
                style={
                  Object {
                    "marginRight": "10px",
                  }
                }
              />
              Add a question
            </button>
          </div>
        </div>
        <br />
        <NewQuestion
          quiz={
            Object {
              "_id": "5eb14db7c8248d0f7cd4a279",
              "quizData": Array [
                Object {
                  "answer": "B",
                  "options": Array [
                    Object {
                      "A": "one",
                    },
                    Object {
                      "B": "two",
                    },
                    Object {
                      "C": "three",
                    },
                  ],
                  "question": "Question 1",
                },
                Object {
                  "answer": "C",
                  "options": Array [
                    Object {
                      "A": "a",
                    },
                    Object {
                      "B": "b",
                    },
                    Object {
                      "C": "c",
                    },
                    Object {
                      "D": "d",
                    },
                  ],
                  "question": "question2",
                },
              ],
              "quizKey": "quiz0",
              "quizName": "First quiz test name",
            }
          }
        />
      </div>
    </form>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
    style={
      Object {
        "margin": "10px",
      }
    }
    type="button"
  >
    <CheckBox
      color="currentColor"
      size="1em"
      style={
        Object {
          "marginRight": "10px",
        }
      }
    />
    Save and Submit
  </button>
</div>
`;

exports[`AddQuizzes should render correctly 1`] = `
<div>
  <h6
    className="container-fluid"
    style={
      Object {
        "margin": 10,
      }
    }
  >
    Hello, here you can create new quizzes
  </h6>
  <a
    href="/quizzes"
  >
    <button
      className="btn btn-primary"
      style={
        Object {
          "margin": "10px",
        }
      }
      type="button"
    >
      <ChevronBarLeft
        color="currentColor"
        size="1em"
        style={
          Object {
            "marginRight": "10px",
          }
        }
      />
      View quizzes
    </button>
  </a>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    style={
      Object {
        "margin": "10px",
      }
    }
    type="button"
  >
    <PlusSquare
      color="currentColor"
      size="1em"
      style={
        Object {
          "marginRight": "10px",
        }
      }
    />
    Add a new Quiz
  </button>
  <br />
</div>
`;
